### PR TITLE
GH-127429: fix sysconfig data generation on cross-builds

### PR DESCRIPTION
**Pull request by FFY00**
* Issue: gh-127429

### CODE REVIEW

**Comment by FFY00** [comment]
@nascheme, this *should* fix the JIT tests.

**Comment by FFY00** [comment]
Alright, the test seem to be passing :smile:

**Comment by FFY00** [comment]
Okay, there seems to be an issue still:

```
======================================================================
ERROR: test_sysconfigdata_json (test.test_sysconfig.TestSysConfig.test_sysconfigdata_json)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/cpython/cpython/Lib/test/test_sysconfig.py", line 644, in test_sysconfigdata_json
    with open(json_data_path) as f:
         ~~~~^^^^^^^^^^^^^^^^
FileNotFoundError: [Errno 2] No such file or directory: '/home/runner/work/cpython/cpython/build/lib.linux-aarch64-3.14-pydebug/_sysconfig_vars_d_linux_aarch64-linux-gnu.json'
----------------------------------------------------------------------
```

**Comment by FFY00** [comment]
Okay, `_PYTHON_SYSCONFIGDATA_PATH` is not being set correctly when generating the file.

```
_PYTHON_HOSTRUNNER='qemu-aarch64' _PYTHON_PROJECT_BASE=/home/runner/work/cpython/cpython _PYTHON_HOST_PLATFORM=linux-aarch64 PYTHONPATH=./Lib _PYTHON_SYSCONFIGDATA_NAME=_sysconfigdata_d_linux_aarch64-linux-gnu _PYTHON_SYSCONFIGDATA_PATH= ../build/bin/python3 -S -m sysconfig --generate-posix-vars ;\
```

**Comment by FFY00** [comment]
It seems like `_get_pybuilddir` was using a runtime check, which reflected the host on cross-builds.

**Comment by FFY00** [comment]
Oh, yay, all tests pass, that wasn't much fun to debug :sweat_smile:

**Comment by hugovk** [comment]
@FFY00 Triage: please could you make the backports if they're still needed? Otherwise let's remove the labels.

File: Lib/sysconfig/__main__.py [hunk 1]
```diff
@@ -161,7 +162,7 @@ def _print_config_dict(d, stream):
 
 def _get_pybuilddir():
     pybuilddir = f'build/lib.{get_platform()}-{get_python_version()}'
-    if hasattr(sys, "gettotalrefcount"):
+    if get_config_var('Py_DEBUG') == '1':
```
Comment (XuehaiPan) [inline Lib/sysconfig/__main__.py:165]: `get_config_var('Py_DEBUG')` is an `int` rather than `str`.

**Pull request merged**